### DISCUSSION TITLE
How to make empty directory tree if not exist using CLI?

### DISCUSSION THREAD

**Discussion by ylazy** [iterate-ch/cyberduck#16714] [Q&A] [answered]
Hi!

I'm using cyberduck to automate some tasks.

I need to make `/ads/v/vietnamicontent/2024/11/18/177415/300x600/test/dfp/pc/std/fixed` on my server
Currently I'm doing like this:

```
duck --mkdir /ads (failed because ads exist)
duck --mkdir /ads/v (success)
duck --mkdir /ads/v/vnfx (success)
duck --mkdir /ads/v/vnfx/2024 (success)
duck --mkdir /ads/v/vnfx/2024/11 (success)
duck --mkdir /ads/v/vnfx/2024/11/18 (success)
duck --mkdir /ads/v/vnfx/2024/11/18/177415 (success)
duck --mkdir /ads/v/vnfx/2024/11/18/177415/300x600 (success)
duck --mkdir /ads/v/vnfx/2024/11/18/177415/300x600/test (success)
duck --mkdir /ads/v/vnfx/2024/11/18/177415/300x600/test/dfp (success)
duck --mkdir /ads/v/vnfx/2024/11/18/177415/300x600/test/dfp/pc (success)
duck --mkdir /ads/v/vnfx/2024/11/18/177415/300x600/test/dfp/pc/std (success)
duck --mkdir /ads/v/vnfx/2024/11/18/177415/300x600/test/dfp/pc/std/fixed (success)
```

The problem is, it's very SLOW. It takes ~70 seconds to create the directory tree above. Duck has to connect/disconnect 13 times to the server to finish the job.

I have also tried `duck --mkdir /ads/v/vnfx/2024/11/18/177415/300x600/test/dfp/pc/std/fixed` to see if I can create multiple directories at once but it failed. I don't know if it was the server's limitation or cyberduck's limitation. Then I tried to create folder by folder (12 times).

Am I missed something? Are there any better ways to do this?

Please help!
Thank you!

**Comment by dkocher** [comment, marked as answer]
We do not currently have a feature to create intermediate directories as required.

**Reply by ylazy** [reply]
yeah... so I'm going to make a feature request...
Thank you!
